### PR TITLE
Remove TiberianSunRefinery 

### LINKED ISSUE #13992: TiberianSunRefinery not needed in the future?
From #13721 :
> And I came to a conclusion that it is the activity that knows what to do when docking, not the client/host actors. For example, if I ordered an MNLY to dock at a FIX it should move to the rally point after reloading and repairing. On the other hand, if it was laying mines and it came back to the dock to reload, it should reload and continue to lay the mines.

If this discussion is accepted, then TiberianSunRefinery should be deleted, as it only provides the dock sequence, which is what VoxelHarvester should know better.

This change will allow mixing voxel and non-voxel harvesters to dock at a refinery.

### PR DESCRIPTION
**Pull request by PunkPun**
Closes #13992
Split from #20168

Supersedes #20184
Partly supersedes #20178

`TiberianSunRefinery` was created as a way for voxel bodies for dock. But It was done in an inefficient and docking refactor blocking way. So in this PR I replace that solution with `IDockClientBody` interface

I also move `WithDockingOverlay` cnc -> common

### CODE REVIEW

File: OpenRA.Mods.Common/UpdateRules/UpdatePath.cs [hunk 1]
```diff
@@ -105,7 +105,7 @@ public class UpdatePath
 				new TextNotificationsDisplayWidgetRemoveTime(),
 				new ExplicitSequenceFilenames(),
 				new RenameEngineerRepair(),
-				new ProductionTabsWidgetAddTabButtonCollection()
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/UpdateRules/UpdatePath.cs:108]: This change seems wrong.

Comment (PunkPun) [reply OpenRA.Mods.Common/UpdateRules/UpdatePath.cs:108]: fixed, something went wrong

File: OpenRA.Mods.Common/Activities/HarvesterDockSequence.cs [hunk 2]
```diff
@@ -110,19 +122,65 @@ public override bool Tick(Actor self)
...
+				});
+			}
+			else
+				PlayUndockClientAnimation(self, () => dockingState = DockingState.Complete);
+		}
 
-		public abstract void OnStateUndock(Actor self);
+		public virtual void PlayUndockClientAnimation(Actor self, Action after)
+		{
+			if (DockClentBody != null)
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Activities/HarvesterDockSequence.cs]:
```suggestion
			if (DockClientBody != null)
```

Comment (PunkPun) [reply OpenRA.Mods.Common/Activities/HarvesterDockSequence.cs]: fixed

**Review by abcdefg30 — APPROVED** [review]

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/1a1de287d92b2dd1d8ebb280d7d6fecc83109c2e...7ada13685369a4b54a20ce8ba0b825ee56bf6565)

**Pull request merged**